src/handler/mod.rs

```diff
@@ -494,14 +501,23 @@ impl Handler {
                     .map_err(|e| RequestError::EncryptionFailed(format!("{e:?}")))?;
                 (packet, false)
             } else {
-                // No session exists, start a new handshake
+                // No session exists, start a new handshake initiating a new session
                 trace!(
                     "Starting session. Sending random packet to: {}",
                     node_address
                 );
                 let packet =
                     Packet::new_random(&self.node_id).map_err(RequestError::EntropyFailure)?;
-                // We are initiating a new session
+                // Queue the request for sending after the handshake completes
+                self.pending_requests
+                    .entry(node_address.clone())
+                    .or_insert_with(Vec::new)
+                    .push(PendingRequest {
+                        contact: contact.clone(),
+                        request_id: request_id.clone(),
+                        request: request.clone(),
+                    });
```

Comment:
I think this is unnecessary. The entry in `pending_requests` inserted here remains despite the request having been completed.

```mermaid

sequenceDiagram
    participant Node1
    participant Node2

    Note over Node1: Send FINDNODE request to Node2.

    rect rgb(10, 10, 10)
    Note left of Node1: Handler::send_request()
    Note over Node1: Send random packet <br> since no session exists.
    Note over Node1: pending_requests.insert
    Note over Node1: active_requests.insert
    end



    Node1 ->> Node2: Random packet
    Node2 ->> Node1: WHOAREYOU

    rect rgb(10, 10, 10)
    Note left of Node1: Handler::handle_challenge()
    Note over Node1: active_requests.remove_by_nonce
    Note over Node1: Generate a new session <br> and create authentication packet
    Note over Node1: active_requests.insert
    end

    Node1 ->> Node2: Handshake message 
    Node2 ->> Node1: NODES

    rect rgb(10, 10, 10)
    Note left of Node1: Handler::handle_response()
    Note over Node1: active_requests.remove_request
    end

    rect rgb(100, 0, 0)
    Note over Node1: The request has completed <br> but the entry in pending_requests remains.
    end

```